src/common/log.ts

```diff
@@ -1,27 +1,167 @@
-export type LogCallback = (message: string) => void;
+import { CLIError } from '@oclif/errors';
+import { green, grey, red, yellow } from 'chalk';
 
-export function formatShellLog(
-  initial: string | undefined,
-  quoted?: string[],
-  env?: Record<string, string>
-): string {
-  let envString = '';
-  if (env !== undefined) {
-    envString =
-      Object.entries(env)
-        .map(([key, value]) => `${key}='${value}'`)
-        .join(' ') + ' ';
+import { messages } from './messages';
+
+interface ILogger {
+  info(input: string): void;
+  warn(input: string): void;
+  success(input: string): void;
+  error(input: string): void;
+}
+class StdoutLogger implements ILogger {
+  public static readonly successPrefix = '🆗';
+  public static readonly errorPrefix = '❌';
+  public static readonly warningPrefix = '⚠️';
```

Comment:
I would consider adding a `--no-emoji` flag or env var for users that don't want emoji in the output or don't have their terminal configured to correctly display it anyway.